Source/DafnyCore/Triggers/QuantifiersCollection.cs

```diff
@@ -338,7 +339,7 @@ internal class QuantifierGroup {
           msg.Append(WARN).AppendLine("Suppressing loops would leave this expression without triggers.");
           FirstLetterCapitalOnNestedToken();
         } else if (suppressWarnings) {
-          errorLevel = ErrorLevel.Warning;
+          errorLevel = options.WarningsAsErrors ? ErrorLevel.Error : ErrorLevel.Warning;
```

Comment:
Isn't this the wrong location to check `options.WarningsAsErrors` ? Why wouldn't the reporter do that check? (as it already mostly does, but only when you call `reporter.Warning`, not when you call `reporter.Message`

Reply:
Agreed. Implemented.